R/food_group_score.R

```diff
@@ -0,0 +1,307 @@
+#'    *meat* | binary variables indicate child get the food group related to flesh foods (meat, fish, poultry, organ meats) at yesterday (meat =1 ) or not (meat = 0)
+#'    *eggs* | binary variables indicate child get the any kind of eggs at yesterday (eggs =1 ) or not (eggs = 0)
+#'    *vita_fruveg* | binary variables indicate child get the food group related to vitamin A-rich fruits and vegetables at yesterday (vita_fruveg =1 ) or not (vita_fruveg)
+#'    *oth_fruveg* | binary variables indicate child get the food group related to other fruits and vegetables at yesterday (oth_fruveg =1 ) or not (oth_fruveg = 0)
+#'    *food_score* | continuous variables indicate child consume number of food groups including solid, semi-solid or soft foods and breastfeeding
+#'
+#'
+#'
+#' @examples
+#'
+#' # Individual Food Group Variable Construction
+#'
+#'  df <- cfData
+#'
+#'  breastmilk <- fg_breastmilk(df$child_bfyest, df$calc_age_months)
+#'
+#'  grains <- fg_grains(df$child_rice, df$child_potatoes, df$calc_age_months)
+#'
+#'  pulses <- fg_pulses(df$child_beans, df$calc_age_months)
+#'
+#'  diary <- fg_milk(df$child_bms, df$child_milk, df$child_mproduct, df$child_yogurt, df$child_yogurt, df$calc_age_months)
+#'
+#'  meat <- fg_meat(df$child_organ, df$child_insects, df$child_beef, df$child_fish, df$calc_age_months)
+#'
+#'  eggs <- fg_egg(df$child_eggs, df$calc_age_months)
+#'
+#'  vita_fruveg <- fg_vita_fruveg(df$child_pumpkin, df$child_leafyveg, df$child_mango, df$calc_age_months)
+#'
+#'  oth_fruveg <- fg_oth_fruveg(df$child_fruit, df$child_fruit, df$calc_age_months)
+#'
+#'
+#'  # Calculate Food Consumption Score
+#'  food_score <- fg_score(breastmilk, grains, pulses, diary, meat,
+#'                         eggs, vita_fruveg, oth_fruveg)
+#'
+#'  df$food_score <- food_score
+#'
+#' @author Nicholus Tint Zaw
+#' @export
+#' @rdname fg_score
+#'
+#'
+#################################################################################
+
+## Food Groups Score ##
+fg_score <- function(breastmilk, grains, pulses, diary, meat, eggs,
+                     vita_fruveg, oth_fruveg){
+
+  if(!is.null(breastmilk) & !is.null(grains) & !is.null(pulses) &
+     !is.null(diary) & !is.null(meat) & !is.null(eggs) &
+     !is.null(vita_fruveg) & !is.null(oth_fruveg)){
+
+       fg_group <- as.data.frame(
+         cbind(breastmilk, grains, pulses, diary, meat, eggs, vita_fruveg, oth_fruveg)
+         )
+
+       food_score <- rowSums(fg_group, na.rm = TRUE)
+
+       return(food_score)
+     }
+}
+
+################################################################################
+################################################################################
+
+## Food Groups Construction ##
+
+################################################################################
+#
+#' @export
+#' @rdname fg_score
+#'
+#
+################################################################################
+
+# Breast milk
+fg_breastmilk <- function(q4, age){
```

Comment:
why do we need the age argument in all these functions here?

is is possible for us to create one universal generic function that can score any combination of foods to be equal to 1 or 0?

Reply:
* `aggregate()` function

x should be a list()

```
x <- list(organ_meat, meat, insect, fish)

score_foods <- function(x) {
  y <- sum(x, na.rm = TRUE)

  if (y > 0) score <- 1 else score <- 0

 score
}
```

Reply:
You don't have to export this function and it can be a miscellaneous function/helper function:

for package users, they need to use this approach: `riycf:::score_foods()`

bur for package developer, they can use this function within the package like this: `scrore_foods()`